librosa/core/spectrum.py

```diff
@@ -283,7 +285,11 @@ def istft(stft_matrix, hop_length=None, win_length=None, window=None,
         spec = np.concatenate((spec.conj(), spec[-2:0:-1]), 0)
         ytmp = ifft_window * fft.ifft(spec).real
 
-        y[sample:(sample+n_fft)] = y[sample:(sample+n_fft)] + ytmp
+        y[sample:(sample + n_fft)] = y[sample:(sample + n_fft)] + ytmp
+        ifft_window_sum[sample:(sample + n_fft)] += ifft_window_square
+
+    # Normalize by window
+    y[ifft_window_sum != 0] /= ifft_window_sum[ifft_window_sum != 0]
```

Comment:
A couple of points here:

1) The index set ``ifft_window_sum != 0`` should only be computed once
2) This normalization isn't numerically stable.  Instead, it should perhaps be replaced by ``ifft_window_sum < util.SMALL_FLOAT``.  (If ``ifft_window_sum`` can go negative, which it shouldn't, replace it by ``np.abs(ifft_window_sum)``.)